test/unit/lib/messages.js

```diff
@@ -60,4 +60,49 @@ describe('messages', () => {
       expect(message).to.equal(`If your surgery is not here, check the postcode and text you have entered are right and ${searchAgainLink}.`);
     });
   });
+
+  describe('no results prompt', () => {
+    const res = {};
+    res.locals = { searchErrorClass: '' };
+
+    it('should have a different message when there are results', () => {
+      const message = messages.noResultsMessage(res, false, 'sheffield', 's3');
+
+      expect(message).to.equal('<h2>Find your GP surgery</h2>  <p>You need to do this to go to your booking system. Enter the name of your surgery, ' +
+      'the name of your GP or a postcode.</p>');
+    });
+
+    it('should have a GP name message when only a name has been passed', () => {
+      const message = messages.noResultsMessage(res, true, undefined, 'netherthorpte');
+
+      expect(message).to.equal('<h2>We can not find a surgery using \'netherthorpte\'</h2> <p>Check the name you entered ' +
+        'is right. You get better results if you enter a full name. Or you can search with a postcode instead.</p>');
```

Comment:
I assume that this text will change in the review with Dean as it talks about 'Or ... postcode'